src/python/pants/backend/python/util_rules/pex_cli.py

```diff
@@ -39,6 +39,7 @@ class PexBinary(TemplatedExternalTool):
 
     default_version = "v2.1.42"
     default_url_template = "https://github.com/pantsbuild/pex/releases/download/{version}/pex"
+    version_constraints = ">=2.1.42, <3.0"
```

Comment:
Not sure what is sensible values here, but it seemed from #12086 that we need at least 2.1.42, and assuming that a future v3 is a potential breaking release, it seems legit to cap at that.